### DISCUSSION TITLE
Triton Intersection

### DISCUSSION THREAD

**Discussion by marcoma9023** [pelicanmapping/osgearth#1859] [Support Q&A]
Hi,

I'm working and integrating Triton SDK into my OSGEarth project...

Looking the example osgearth_triton, into the addBuyancyTest method:
`
isect->setAnchor(anchor);

        // generate a bunch of local points around the anchor:
        for(int x=-50; x<=50; x+=25)
        {
            for(int y=-50; y<=50; y+=25)
            {
                isect->addLocalPoint(osg::Vec3d(x, y, 0));
            }
        }
`

where anchor is the geopoint.

Then, I'm trying to integrate the same logic into the project. 
Then, with `isect->setAnchor(geopoint)` I put the geopoint of my model.

The problem is that I did not understand, to method `addLocalPoint` which points I have to pass?

For example, I have width and length of the model, then I calculated the geopoint for the vertices of the models, how should I convert them to insert them in the `addLocalPoint` method?

Thanks

**Comment by gwaldron** [comment]
The idea is that, for a model like a ship, it is ideal to "clamp" it to the ocean at multiple points. If you only clamp it at the center point of the ship, it will look unnatural as it bobs straight up and down on a wave. If you clamp it at the bow and the stern, it looks better, and better still if you have 3 clamping points, say one in the bow and two on either side of the stern.

So you can set you anchor point anywhere and then apply the local offset points to represent the clamping locations relative to the anchor point (in meters). At least one local point is required -- you can put it at (0,0) if you just want to clamp the center point. I hope that makes sense!

(You can see the effect by altering the `osgearth_triton` demo to use only one or 2 local points.)

**Reply by marcoma9023** [reply]
I proceeded by adding different points as local points to build the shape of the ship.

Then, since I needed to have a smooth effect, I averaged the x, y and z components of the normals returned by the TritonIntersections..

Thanks

**Comment by marcoma9023** [comment]
However, if I want to remove dynamically a TritonIntersections?

Because in the OsgEarth source code, in the TritonLayer.cpp I added the method that remove from the TritonLayer all the intersections:

`void
TritonLayer::removeIntersections(TritonIntersections* value) {
	TritonLayerNode* node = static_cast<TritonLayerNode*>(_tritonNode.get());
	if (node != nullptr) {
		node->_isect.clear();
}`

but this causes the osgearth dll to crash. Why?
